### PR TITLE
feat: implement no-test-isolation rule to prevent test interdependencies

### PR DESCRIPTION
**Pull request by tigredonorte**
## Summary

This PR implements the `no-test-isolation` rule to help prevent test flakiness by detecting patterns that can cause tests to affect each other's state.

## What This Rule Does

The rule identifies common anti-patterns that lead to test interdependencies:
- **Shared mutable state across tests** - Detects when tests share variables that could leak state
- **Missing cleanup in afterEach hooks** - Ensures proper teardown after each test
- **Direct database/API modifications without isolation** - Catches unprotected external state changes
- **Global variable mutations** - Identifies changes to global state that persist between tests

## Features

✅ **Auto-fixable** - Provides automatic fixes where possible
✅ **Comprehensive detection** - Covers multiple test isolation anti-patterns
✅ **Clear error messages** - Detailed guidance to help developers fix issues
✅ **Framework support** - Works with Jest, Mocha, and other popular test frameworks

## Files Changed

- `lib/rules/no-test-isolation.js` - Core rule implementation
- `tests/lib/rules/no-test-isolation.test.js` - Comprehensive test suite
- `docs/rules/no-test-isolation.md` - Rule documentation with examples
- `examples/no-test-isolation-violations.test.js` - Example violations for testing

## Testing

The rule has been thoroughly tested with:
- Valid test patterns that maintain proper isolation
- Invalid patterns covering all detection scenarios
- Auto-fix verification for fixable violations
- Edge cases and framework-specific patterns

## Example

```javascript
// ❌ Bad - Shared state between tests
let sharedCounter = 0;
test('first test', () => {
  sharedCounter++;
  expect(sharedCounter).toBe(1);
});

// ✅ Good - Isolated state
test('first test', () => {
  let localCounter = 0;
  localCounter++;
  expect(localCounter).toBe(1);
});
```

### CODE REVIEW

File: lib/rules/no-test-isolation.js [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+          // Also check for non-mutating method calls on shared objects (still problematic for isolation)
+          if ((moduleVariables.has(objectName) || describeVariables.has(objectName)) &&
+              !mutatingMethods.includes(method) && method !== 'toString' && method !== 'valueOf') {
+            // Check if this is a meaningful method call that could affect test isolation
+            const meaningfulMethods = [
+              'method', 'call', 'apply', 'bind', 'run', 'execute', 'perform', 'process',
+              'start', 'stop', 'close', 'open', 'connect', 'disconnect', 'send', 'receive'
+            ];
+
+            if (meaningfulMethods.includes(method) || method.match(/^[a-z]/)) {
```
[the remark this answers is not in the document]
Comment (tigredonorte) [reply lib/rules/no-test-isolation.js]: This is incorrect. The regex is **already defined as a constant** `LOWERCASE_START_REGEX` at line 10 of the file:

```javascript
const LOWERCASE_START_REGEX = /^[a-z]/;
```

And it's used at line 537:
```javascript
if (meaningfulMethods.includes(method) || LOWERCASE_START_REGEX.test(method))
```

No optimization needed - it's already optimized.
